internal/manager/run.go

```diff
@@ -110,6 +110,13 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic, d
 
 	kongSemVersion := semver.Version{Major: v.Major(), Minor: v.Minor(), Patch: v.Patch()}
 
+	if kongSemVersion.LT(versions.KICv3VersionCutoff) {
+		return fmt.Errorf(
+			"Kong Gateway version: %q is not supported by Kong Kubernetes Ingress Controller in version >=3.0.0, the lowest supported version is: %q",
+			kongSemVersion, versions.KICv3VersionCutoff,
+		)
+	}
```

Comment:
That's done only once at startup -- I wonder if we should also validate this when discovering gateways. This way we could filter out gateways that do not satisfy version requirements.

Reply:
Good point, I can leave the above and add a check for discovering gateways too. In such cases log a WARN message that this Gateway will be ignored